src/qibo/optimizers/abstract.py

```diff
@@ -0,0 +1,92 @@
+import inspect
+from abc import abstractmethod
+
+import numpy as np
+
+from qibo.config import raise_error
+
+
+def check_options(function, options):
+    """
+    Check if given options dictionary is compatible with optimization
+    method arguments.
+    """
+
+    for arg in options:
+        if arg not in list(inspect.signature(function).parameters):
+            raise_error(
+                TypeError,
+                f"Given argument {arg} is not accepted by {function.__code__.co_name} function, which is the fitting function of the chosen optimizer.",
+            )
+
```

Comment:
This checking function is useful if we keep the dictionaries approach, because a dictionary can be generally updated with any key, but then we should check if the keys are accepted by the optimization function or not. 
I had a discussion about dictionaries with both @alecandido and @andrea-pasquale. Here there is a conflict between the flexibility I want to guarantee and the complexity of doing this.

A solution can be to use the safer `dataclasses`, or to constrain the optimizers arguments to the ones we want to accept, but by doing this we will loose the flexibility of the interface I am proposing. 

Maybe we can discuss a little bit during the Qibo meeting :)

Reply:
My suggestion would have been to pass all arguments down below. If it's not accepted by optimization, the optimization itself will fail ^^